service-commands/scripts/setup-instance.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Usage: setup.sh [options] <service> <name>
+#
+# Options:
+#   -p <provider>
+#   -u <user>
+#   -c <audius-k8s-manifests-config>
+#   -r <audius-protocol-git-ref>
+#   -l <audius-client-git-ref>
+#   -f
+
+set -e
+
+PROTOCOL_DIR=${PROTOCOL_DIR:-$(dirname $(realpath $0))/../../}
+
+source $PROTOCOL_DIR/service-commands/scripts/utils.sh
+
+# parse arguments
+while getopts "p:u:c:r:l:f" flag; do
+	case "$flag" in
+		p) provider=$OPTARG;;
+		u) user=$OPTARG;;
+		c) audius_k8_manifests_config=$OPTARG;;
+		r) audius_protocol_git_ref=$OPTARG;;
+		l) audius_client_git_ref=$OPTARG;;
+		f) image=${GCP_DEV_IMAGE};;
+	esac
+done
+
+service=${@:$OPTIND:1}
+name=${@:$OPTIND+1:1}
+
+# Set defaults and validate arguments
+provider=${provider:-$DEFAULT_PROVIDER}
+user=${user:-$DEFAULT_USER}
+audius_protocol_git_ref=${user:-$DEFAULT_AUDIUS_PROTOCOL_GIT_REF}
+audius_client_git_ref=${user:-$DEFAULT_AUDIUS_CLIENT_GIT_REF}
+
+if [[ "$provider" != "gcp" ]] && [[ "$provider" != "azure" ]]; then
+	echo "Unknown provider:" $provider
+	exit 1
+fi
+
+if [[ "$service" != "creator-node" ]] && [[ "$service" != "discovery-provider" ]] && [[ "$service" != "remote-dev" ]]; then
+	echo "Unknown service:" $service
+	exit 1
+fi
+
+if [[ -z "$name" ]]; then
+	echo "Name of instance was not provided"
+	exit 1
+fi
+
+# Check dependencies
+case "$provider" in
+	azure)
+		if [[ ! $(type az 2> /dev/null) ]]; then
+			echo "az: not found"
+			echo "aborting"
+			exit 1
+		fi
+		azure_setup_defaults
+		;;
+	gcp)
+		if [[ ! $(type gcloud 2> /dev/null) ]]; then
+			echo "gcloud: not found"
+			echo "aborting"
+			exit 1
+		fi
+		gcp_set_defaults
+		;;
+esac
+
+# Create instance if it does not exist
```

Comment:
instead of comment-headed blocks, these could be factored into named bash fns - but ymmv here!

Reply:
I agree, but I think this can be saved for a future PR